pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -11,413 +11,415 @@
+from pytorch_lightning.tuner.auto_gpu_select import pick_multiple_gpus
+from pytorch_lightning.utilities import AMPType, _NATIVE_AMP_AVAILABLE, _APEX_AVAILABLE, device_parser
+from pytorch_lightning.utilities import rank_zero_only
+from pytorch_lightning.utilities.distributed import rank_zero_warn, rank_zero_info
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning import _logger as log
 from pytorch_lightning.cluster_environments.slurm_environment import SLURMEnvironment
 from pytorch_lightning.cluster_environments.torchelastic_environment import TorchElasticEnvironment
-from pytorch_lightning.utilities import device_parser, rank_zero_only, _TPU_AVAILABLE
-from pytorch_lightning.utilities.distributed import rank_zero_info, rank_zero_warn
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-
-if _HOROVOD_AVAILABLE:
-    import horovod.torch as hvd
 
+try:
+    import torch_xla
+except ImportError:
+    XLA_AVAILABLE = False
+else:
+    XLA_AVAILABLE = True
 
-class AcceleratorConnector:
-
-    def __init__(self, trainer):
-        self.trainer = trainer
-        self.accelerator = None
-
-    def on_trainer_init(
-            self,
-            num_processes,
-            tpu_cores,
-            accelerator,
-            distributed_backend,
-            auto_select_gpus,
-            gpus,
-            num_nodes,
-            log_gpu_memory,
-            sync_batchnorm,
-            benchmark,
-            replace_sampler_ddp,
-            deterministic,
+try:
+    import horovod.torch as hvd
+except (ModuleNotFoundError, ImportError):
+    _HOROVOD_AVAILABLE = False
+else:
+    _HOROVOD_AVAILABLE = True
+
+
+class BackendConnector(object):
+    def __init__(
+        self,
+        num_processes,
+        tpu_cores,
+        distributed_backend,
+        auto_select_gpus,
+        gpus,
+        num_nodes,
+        sync_batchnorm,
+        benchmark,
+        replace_sampler_ddp,
+        deterministic,
+        precision,
+        amp_type, 
+        amp_level,
+        cluster_environment,
     ):
-        # temp until we remove all dist backend references
-        distributed_backend = self._map_deprecated_dist_backend(accelerator, distributed_backend)
-
-        self.trainer.deterministic = deterministic
 
-        torch.backends.cudnn.deterministic = self.trainer.deterministic
-        if self.trainer.deterministic:
-            # fixing non-deterministic part of horovod
-            # https://github.com/PyTorchLightning/pytorch-lightning/pull/1572/files#r420279383
-            os.environ["HOROVOD_FUSION_THRESHOLD"] = str(0)
-
-        # distributed backend choice
-        self.trainer.distributed_backend = distributed_backend.lower() if distributed_backend else None
+        # initialization
+        self.use_dp = False
+        self.use_ddp = False
+        self.use_ddp2 = False
+        self.use_horovod = False
+        self.use_single_gpu = False
```

Comment:
this is refactored to `_distrib_type` but we can rename it if you want...

Reply:
Yes I know, we need to update the branch with these changes. It's a bit challenging now with keeping uptodate. But we will keep it in mind thanks

Reply:
probably no need for an update as you will cut this work to multiple PRs, juts for you to be aware of it... :]